### PR TITLE
Support file uploads

### PR DESCRIPTION
**Pull request by felipeerias**
This PR implements file uploads (see [`<input type="file">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file)).

When the user activates that functionality, we display a dialog to pick between the files that have been previously downloaded by Wolvic. In this first version, we don't support uploading files that come from other apps in the system.

We support the [`accept`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept) attribute, which lists the file types that the file input should accept.

We also support the [`multiple`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#multiple) attribute, which allows the user to select and upload more than one file.

Fixes https://github.com/Igalia/wolvic/issues/208

### CODE REVIEW

**Comment by felipeerias** [comment]
Test page: http://people.igalia.com/femorandeira/input/

File: app/src/common/shared/com/igalia/wolvic/ui/adapters/ThumbnailAsyncTask.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    private static final int DEFAULT_SIZE_KIND = MediaStore.Images.Thumbnails.MICRO_KIND;
+
+    private final ContentResolver mContentResolver;
+    private final Uri mFileUri;
+    private final OnSuccessDelegate mOnSuccessDelegate;
+    private CancellationSignal mCancellationSignal;
+
+    public ThumbnailAsyncTask(@NonNull Context context, @NonNull Uri fileUri, OnSuccessDelegate onSuccessDelegate) {
+
+        Log.e(LOGTAG, "new ThumbnailAsyncTask");
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/adapters/ThumbnailAsyncTask.java:44]: Nit. `Log.d`. Same for most (some are real errors) logging statements in this file.

File: app/src/common/shared/com/igalia/wolvic/ui/adapters/ThumbnailAsyncTask.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            if (mimeFromExtension.startsWith("audio")) {
+                Log.e(LOGTAG, "  createAudioThumbnail");
+                return ThumbnailUtils.createAudioThumbnail(file, DEFAULT_SIZE, mCancellationSignal);
+            } else if (mimeFromExtension.startsWith("video")) {
+                Log.e(LOGTAG, "  createVideoThumbnail");
+                return ThumbnailUtils.createVideoThumbnail(file, DEFAULT_SIZE, mCancellationSignal);
+            } else if (mimeFromExtension.startsWith("image")) {
+                Log.e(LOGTAG, "  createImageThumbnail");
+                return ThumbnailUtils.createImageThumbnail(file, DEFAULT_SIZE, mCancellationSignal);
+            }
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/adapters/ThumbnailAsyncTask.java:94]: Unftorunate that there isn't a thumbnailutils method that takes the mime type and calls the appropriate method

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/prompts/FilePromptWidget.java [hunk 3]
```diff
@@ -82,12 +81,14 @@ protected void initialize(Context aContext) {
...
 
             mPromptDelegate.dismiss();
+            hide(REMOVE_WIDGET);
         });
 
         mUploadButton = findViewById(R.id.positiveButton);
         mUploadButton.setOnClickListener(view -> {
             // TODO upload the file
             mPromptDelegate.dismiss();
+            hide(REMOVE_WIDGET);
```
Comment (svillar) [inline app/src/common/shared/com/igalia/wolvic/ui/widgets/prompts/FilePromptWidget.java:101]: What are those for?

**Review by svillar — CHANGES_REQUESTED** [review]
Looks pretty good.

I just have a comment regarding visuals. Can we make the buttons in the dialog like the buttons in the setttings widgets? Also the fonts look pretty different. If it's very difficult we can land this as is, as do it in a follow-up but if it's easy let's do it now.

File: app/src/common/shared/com/igalia/wolvic/ui/widgets/prompts/FilePromptWidget.java (hunk 3 above)
Comment (felipeerias) [reply app/src/common/shared/com/igalia/wolvic/ui/widgets/prompts/FilePromptWidget.java:101]: `hide(REMOVE_WIDGET)` closes and removes this dialog.

File: app/src/common/shared/com/igalia/wolvic/ui/adapters/ThumbnailAsyncTask.java (hunk 2 above)
Comment (felipeerias) [reply app/src/common/shared/com/igalia/wolvic/ui/adapters/ThumbnailAsyncTask.java:94]: Yes, that's true.

File: app/src/common/shared/com/igalia/wolvic/ui/adapters/ThumbnailAsyncTask.java (hunk 1 above)
Comment (felipeerias) [reply app/src/common/shared/com/igalia/wolvic/ui/adapters/ThumbnailAsyncTask.java:44]: Fixed.

**Review by svillar — APPROVED** [review]
I'm loving it. Thanks!

**Pull request merged**